### PR TITLE
pinnwand: init at 1.1.0

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

> A Python pastebin that tries to keep it simple. https://supakeen.com/project/pinnwand

https://github.com/supakeen/pinnwand

This is the pastebin software used at https://bpa.st/.

Had to <s>introduce `tornado_6` and<&s> package `pygments-better-html` (has no tests) as dependencies.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pygments-better-html/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  propagatedBuildInputs = [ pygments ];
+
+  # has no tests
+  doCheck = false;
+
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/Kwpolska/pygments_better_html";
+    description = "Improved line numbering for Pygments’ HTML formatter.";
+    license = licenses.bsd3;
+  };
```
Comment (jonringer) [inline pkgs/development/python-modules/pygments-better-html/default.nix:27]: please add yourself as a maintainer

File: pkgs/development/python-modules/pygments-better-html/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "028szd3k295yhz943bj19i4kx6f0pfh1fd2q14id0g84dl4i49dm";
+  };
+
+  propagatedBuildInputs = [ pygments ];
+
+  # has no tests
+  doCheck = false;
+
```
Comment (jonringer) [inline pkgs/development/python-modules/pygments-better-html/default.nix:22]: If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors.

**Review by jonringer — COMMENTED** [review]
otherwise LGTM

File: pkgs/development/python-modules/pygments-better-html/default.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/development/python-modules/pygments-better-html/default.nix]: Done.

File: pkgs/development/python-modules/pygments-better-html/default.nix (hunk 2 above)
Comment (mweinelt) [reply pkgs/development/python-modules/pygments-better-html/default.nix:22]: Done.

**Comment by jonringer** [comment]
is this the only package which needs tornado 6? I would prefer to remove pinning from python_modules, as it can make incoherent environments.

please move the pinning into the application expressions, aws-cli is an example of how to do this

**Comment by mweinelt** [comment]
Yeah, currently this is the only package needing tornado 6, so I added the overwrite into the application expression.

File: pkgs/servers/pinnwand/default.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  ];
+
+  # tests are only available when fetching from GitHub, where they in turn don't have a setup.py :(
+  doCheck = false;
+
+  checkInputs = [ pytest ];
+  checkPhase = ''
+    # https://github.com/supakeen/pinnwand/blob/master/.pre-commit-config.yaml
+    python -m pytest -q
+  '';
```
Comment (jonringer) [inline pkgs/servers/pinnwand/default.nix:40]:
```suggestion
  # tests are only available when fetching from GitHub, where they in turn don't have a setup.py :(
  # ensure that executable is able to import dependencies correctly
  checkPhase = ''
    $out/bin/pinnwand --help > /dev/null
  '';
```

Comment (mweinelt) [reply pkgs/servers/pinnwand/default.nix]: Done, good call.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 88995 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>3 packages built:</summary><br>- pinnwand<br>- python37Packages.pygments-better-html<br>- python38Packages.pygments-better-html</details>

**Pull request merged**